aws_lambda_powertools/shared/headers_serializer.py

```diff
@@ -65,8 +66,9 @@ def serialize(self, headers: Dict[str, Union[str, List[str]]], cookies: List[Coo
             if isinstance(values, str):
                 payload[key].append(values)
             else:
-                for value in values:
-                    payload[key].append(value)
+                if values:
+                    for value in values:
+                        payload[key].append(value)
```

Comment:
we can drop an additional for loop by using `list.extend`, as it accepts an Iterable and it'll add all elements at the end of the list. This drops the complexity